### PR TITLE
Rework form legal section

### LINKED ISSUE #10: add checkbox to allow sharing data with sponsors
?

### LINKED ISSUE #12: Split MLH code of conduct and terms of service/privacy policy
As requested, have three separate checkboxes for CoC, ToS/PP, and data share consent.

### PR DESCRIPTION
**Pull request by ViRb3**
Closes #10 and closes #12.

### CODE REVIEW

File: applications/forms.py [hunk 1]
```diff
@@ -56,24 +56,30 @@ class ApplicationForm(OverwriteOnlyModelFormMixin, BetterModelForm):
...
+    data_consent = forms.BooleanField(required=False,
+                                      label="I authorize you to share my application/registration information "
+                                      "for event administration, ranking, MLH administration, pre- and "
+                                      "post-event informational emails, and occasional messages about "
+                                      "hackathons in-line with the <a href='https://mlh.io/privacy' "
+                                      "target='_blank'>MLH Privacy Policy</a>.", )
+
+    sponsor_consent = forms.BooleanField(required=False,
+                                         label="I authorize you to share my resume, uploaded as part "
+                                         "of this application, with sponsors of the event.", )
```
Comment (elenalape) [inline applications/forms.py]: Mmm, I'd rephrase to "I would like my CV to be shared with the event sponsors"

**Review by elenalape — CHANGES_REQUESTED** [review]

File: applications/forms.py (hunk 1 above)
Comment (qaisjp) [reply applications/forms.py]: How about `I would like my CV, name, and year of graduation to be shared with the event sponsors`?

Comment (ViRb3) [reply applications/forms.py]: Implemented.

**Comment by ViRb3** [comment]
Tested and working, feel free to merge.

**Comment by qaisjp** [comment]
Run makemigrations and add resultant migration pls

**Comment by ViRb3** [comment]
```
Migrations for 'applications':
  applications/migrations/0015_auto_20190213_2347.py:
    - Alter field description on application
    - Alter field diet on application
    - Alter field projects on application
    - Alter field tshirt_size on application
```

**Comment by qaisjp** [comment]
hmm, seems a lot more than needed.

try first checking out master, reset docker, do the regular `migrate`, then run makemigrations (should say "no migrations needed"), then checkout `legal`, and run makemigrations again.

it should generate a file with only the relevant changes (you should rename the resultant migration file).

An example PR that includes migrations can be found in the [base repository](https://github.com/HackAssistant/registration), issue 231. (Not linking directly as I don't want it to show up as a cross-reference).

**Comment by ViRb3** [comment]
It appears to say that no migrations are needed even after switching to the legal branch.

**Comment by qaisjp** [comment]
The problem was that models.py was not being updated, meaning the db was not being updated (i.e, those two optional checkboxes were just being thrown away).

It's not possible to easily have true/false checkboxes in Django (checking the box = true, unchecking the box = null, for some reason); so I have had to make it look like this

image

(this screenshot is on the edit application form only, dw, the rest is still on the main form)

`makemigrations` will now work, but lets forget about migrations, as it will clash with upstream.

**Review by qaisjp — APPROVED** [review]
